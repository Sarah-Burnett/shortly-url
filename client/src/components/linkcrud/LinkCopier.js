import React, { useContext } from 'react';
import { Button, PrimaryButton, LinkModal, darkViolet } from '../../Styles';
import { LinkUpdateContext } from '../../context/LinkUpdateContext';
import { useCopyLink } from '../../utilities/useCopyLink';
import styled from 'styled-components';
import { shortlyUrl } from '../../utilities/url';


const DeactivatedButton = styled(PrimaryButton)`
    background: ${darkViolet};
`

export const LinkCopier = () => {
    const [ linkUpdate, setLinkUpdate ] = useContext(LinkUpdateContext);
    const { link } = linkUpdate;
    const [ copiedLink, copyLink ] = useCopyLink();
    return (
            <LinkModal>
                    <h3>Copy Shortly Link</h3>
                    <input value={shortlyUrl + link.short} readOnly/>
                    { !copiedLink && <PrimaryButton type="button" id="copy" onClick={event => copyLink(event, false)}>Copy</PrimaryButton> }
                    { copiedLink && <DeactivatedButton type="button" id="copy" disabled>Copied</DeactivatedButton> }
                    <Button type="button" onClick={() => setLinkUpdate('')}>Close</Button>
            </LinkModal> 
    )
}
