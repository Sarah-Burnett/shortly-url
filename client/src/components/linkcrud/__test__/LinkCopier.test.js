import {
	rendersWithoutCrashing,
	matchesSnapshot,
	ComponentWrapper,
} from "../../../setupTests";
import LinkCopier from "../LinkCopier";

const LinkCopierWrapper = () => ComponentWrapper(LinkCopier);

it("renders link copier without crashing", () =>
	rendersWithoutCrashing(LinkCopierWrapper));

it("link copier matches snapshot", () => matchesSnapshot(LinkCopierWrapper));

//TODO: check copy link copy links - integration
