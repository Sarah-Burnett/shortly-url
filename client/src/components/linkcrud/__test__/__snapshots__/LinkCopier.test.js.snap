// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`link copier matches snapshot 1`] = `
<form
  className="sc-fzozJi blHhKY"
>
  <h3>
    Copy Shortly Link
  </h3>
  <input
    readOnly={true}
    value="https://short.ly/undefined"
  />
  <button
    className="sc-AxirZ dVJbcR"
    id="copy"
    onClick={[Function]}
    type="button"
  >
    Copy
  </button>
  <button
    className="sc-AxjAm kxpUMR"
    onClick={[Function]}
    type="button"
  >
    Close
  </button>
</form>
`;
